cms/djangoapps/contentstore/tests/test_libraries.py

```diff
@@ -458,26 +495,6 @@ def test_refreshes_children_if_capa_type_change(self):
         html_block = modulestore().get_item(lc_block.children[0])
         self.assertEqual(html_block.display_name, name2)
 
-    def test_refresh_fails_for_unknown_library(self):
-        """ Tests that refresh children fails if unknown library is configured """
-        # Create a course:
-        with modulestore().default_store(ModuleStoreEnum.Type.split):
-            course = CourseFactory.create()
-
-        # Add a LibraryContent block to the course:
-        lc_block = self._add_library_content_block(course, self.lib_key)
-        lc_block = self._refresh_children(lc_block)
-        self.assertEqual(len(lc_block.children), 0)
-
-        # Now, change the block settings to have an invalid library key:
-        resp = self._update_block(
-            lc_block.location,
-            {"source_library_id": "library-v1:NOT+FOUND"},
-        )
-        self.assertEqual(resp.status_code, 200)
-        with self.assertRaises(ValueError):
-            self._refresh_children(lc_block, status_code_expected=400)
-
```

Comment:
This test, which was failing, heavily overlapped with test_change_after_sync. In fixing it, I found it easiest to smash them together into one big test: test_switch_to_unknown_source_library_preserves_settings.